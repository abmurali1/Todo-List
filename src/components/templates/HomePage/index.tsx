import React, {FC} from 'react';
import {SafeAreaView, StyleSheet, Text, Image, View} from 'react-native';

import Button from '../../atoms/Button/index';
import I18n from '../../../localization/index';

interface HomePageProps {
  title: string;
  navigation: any;
}
const HomePage: FC<HomePageProps> = props => {
  return (
    <SafeAreaView style={{backgroundColor: 'lightgreen', flex: 1}}>
      <View style={styles.container}>
        <Text style={styles.textstyle}>{I18n.t('crypto_market')}</Text>
        <Image
          source={require('../../../assets/images/home/bag.png')}
          style={{height: 300, width: 300}}
        />
        <Button
          title={I18n.t('start')}
          onPress={() => {
            props.navigation.navigate('CoinMarket');
          }}
        />
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  textstyle: {
    textAlign: 'center',
    fontSize: 18,
    fontFamily: 'Arial',
    fontWeight: 'bold',
    color: '#1b1f16',
  },
  container: {flex: 1, alignItems: 'center', justifyContent: 'space-between'},
});

export default HomePage;
