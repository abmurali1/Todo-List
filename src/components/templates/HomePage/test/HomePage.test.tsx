import React from 'react';
import renderer from 'react-test-renderer';
import I18n from '../../../../localization/index';
import HomePage from '../index';

jest.mock('react-native-localize', () => {
  return {
    getLocales: jest.fn(),
  };
});

describe('<Home>', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<HomePage title={I18n.t('start')} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
