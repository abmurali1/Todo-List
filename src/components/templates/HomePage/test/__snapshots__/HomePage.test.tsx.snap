// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "lightgreen",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1b1f16",
          "fontFamily": "Arial",
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Crypto Market
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/home/bag.png",
        }
      }
      style={
        Object {
          "height": 300,
          "width": 300,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#465c47",
          "borderRadius": 10,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": 200,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "#ccb866",
            "fontFamily": "Verdana",
            "fontSize": 20,
            "letterSpacing": 3,
          }
        }
      >
        START
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
